File: ise-yen/PGS/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for(int l = 0; l < commands.size(); l++){
        vector<int> c = commands[l];
        int i = c[0] - 1;
        int j = c[1] - 1;
        int k = c[2] - 1;
        
        vector<int> v;
        while(i <= j){
            v.push_back(array[i]);
            i++;
        }
        
        sort(v.begin(), v.end());
        answer.push_back(v[k]);
    }
    return answer;
}

File: Sjaize/Sorting/K번째수.java
// https://school.programmers.co.kr/learn/courses/30/lessons/42748
import java.util.Arrays;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        for (int index = 0; index < answer.length; index++) {
            int i = commands[index][0] - 1;
            int j = commands[index][1];
            int k = commands[index][2] - 1;
            int[] sub = Arrays.copyOfRange(array, i , j);
            Arrays.sort(sub);
            answer[index] = sub[k];
        }
        
        return answer;
    }
}
